x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/indexlifecycle/PhaseCompleteStep.java

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.core.indexlifecycle;
+
+/**
+ * This is essentially a marker that a phase has ended, and we need to check
+ * the age of an index before proceeding to the next phase.
+ */
+public class PhaseCompleteStep extends Step {
+    public static final String NAME = "complete";
+
+    public PhaseCompleteStep(StepKey key, StepKey nextStepKey) {
+        super(key, nextStepKey);
```

Comment:
I think we need to make the next step key always `null` in these steps? The reason is that the next step after this phase could change before we get there. I think we need to determine the next phase dynamically when we find ourselves at a `PhaseCompleteStep` and have each phase separated from each other in the steps chain. I think this might mean having something like a `MoveToPhaseUpdateTask` so these checks happen on the cluster state update thread where we know the cluster state is static and we can inspect the policy.

I am ok with this happening as a follow up PR though so this PR is kept more contained.

Reply:
I agree with the eventual plan, and I do think this should be a follow up to this PR, especially with @talevy's #33289 which changes how we computer phase steps